### ESMA 3016
### Edgar Acuna
### Lab21: Uso del modulo statsmodels para Inferencia.  Prueba z para  proporciones. 

In [32]:
import numpy as np
import statsmodels.api as sm
import statsmodels.stats.api as sms
import statsmodels.formula.api as smf
from statsmodels.stats.proportion import proportions_ztest
import matplotlib.pyplot as plt
%matplotlib inline

In [33]:
#Intervalo de confianza para la media
#Ejemplo: Intervalo de confianza para la media poblacional del colesterol de todos
# los pacientes con problemas cardiacos asumiendo que su distribucion es  normal
#La varianza (o desviacion estandar) debe ser conocida. En este caso se asume que 
#la desviacion estandar (sigma) es 13. Se usa la muestra de 20 pacientes
colest=[217,223,225,245,238,216,217,226,202,233,235,242,219,221,234,199,236,248,218,224] 
sm.stats.zconfint(colest,alpha=.10)

(221.08387591282869, 230.71612408717132)

In [34]:
#prueba de hipotesis
sm.stats.ztest(colest, x2=None, value=220, alternative='larger', usevar='pooled', ddof=1.0)

(2.0150303902808271, 0.021950732457882954)

#### $H_o: \mu=220$
#### $H_a: \mu>220$

In [35]:
#Caso 2: Poblacion Normal y varianza (sigma^2) desconocida,muestra pequena#Intervalo de confianza para la media
#La media muestral xbarra es un estimador de la media pobacional mu
#Intervalo de confianza para mu del 100(1-alfa)%, alfa es el nivel de significacion
#xbar-t(alfa/2,n-1)*s/sqrt(n),xbarra+t(alfa/2,n-1)*s/sqrt(n)
#Ejemplo 1: Hallar el intervalo de confianza del 95% para el promedio de todos los graduandos
#basado en la siguiente muestra de 14 graduandos
#Asumir que la poblacion es normal 
gpa=[3.15,2.80,2.56,3.17,3.73,2.77,3.18,3.12,3.33,2.99,2.71,2.83,3.45,3.55]

In [36]:
sms.DescrStatsW(gpa).tconfint_mean()

(2.8985215652074281, 3.2929070062211441)

In [37]:
#Ejemplo 7.5. Los tiempos de sobrevivencia (en años) de 12 personas que se han sometido a un transplante de
#corazón son los siguientes:
surv=[3.1,.9,2.8,4.3,.6,1.4,5.8,9.9,6.3,10.4,0,11.5]

In [38]:
#Hallar un intervalo de confianza del 99 por ciento para el promedio de vida
#de todas las personas que se han sometido a un transplante de corazón.
sms.DescrStatsW(surv).tconfint_mean(alpha=.01)

(1.122494220344997, 8.3775057796550048)

In [39]:
#Probar la hipotesis de que el tiempo de sobrevivencia promedio es mas de 4
sms.DescrStatsW(surv).ttest_mean(4,"larger")   

(0.64213678114039097, 0.26696901006689233, 11.0)

#### El valor de la prueba estadistica es  0.64
#### El p-value es 0.2669
#### Los grado de libertad de la prueba t son 11

In [67]:
#Ejemplo 7.7. En 1990 en un cierto país, se reportó que dos de cada 5 personas pensaban que
#debería incrementarse el poder nuclear.  En una encuesta reciente hecha en
#1996 a  1225 personas se encontró que 478 de ellos pensaban que se debería
#aumentar el poder nuclear.  ¿Piensa Ud. que hay evidencia de que
#la opinión de la gente en 1996 ha cambiado con respecto a 1990? Justificar su contestación.
proportions_ztest(478,1225,.4)

(-0.70287005517582912, 0.48213673413505531)

#### La prueba estadistica de Z es -0.702 y el pvalue es 0.4821. No se rechaza la hipotesis Nula

In [71]:
#Ejemplo 7.8. El director de un hospital afirma que el 25 por ciento de los nacimientos que
#ocurren allí son por cesárea.  Un médico que trabaja en dicho hospital piensa
#que ese porcentaje es mayor.  Para probar su afirmación recolecta información
#de los 25 nacimientos ocurridos durante una semana. 
partos=['cesarea','normal','cesarea','normal','normal','normal','normal','cesarea','normal','cesarea',
        'normal','cesarea','normal','normal','normal','normal','normal','cesarea','normal','normal',
        'cesarea','normal','normal','cesarea','normal']

In [72]:
conteos=partos.count('cesarea')

In [73]:
trials=len(partos)

In [74]:
proportions_ztest(conteos,trials,value=.25,alternative='larger')

(0.75030630999847625, 0.22653512156653899)

#### La pueba estadistica aproximada es 0.75 y el p-value es 0.226. No se rechaza la Hipotesis Nula